Test cases for Package_Item(Model)

Jeshan7 opened the pull request:
## Why?
    - Coverage of package_item tests were low 

## This change addresses the need by:
   - Updated the test cases in test/data/model/package_item_test.exs
   - Wrote new test cases in test/data/model/package_item_test.exs

[delivers #163248592]

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

Jeshan7 committed:
[#163248592]
   - Updated the test cases in model/package_item_test.exs
   - Wrote new test cases in model/package_item_test.exs

arjun289 commented on apps/snitch_core/test/data/model/package_item_test.exs:
```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
-      assert {:error, changeset} =
+    test "successful with valid attributes", context do
+      %{package_item: package_item} = new_package(context)
+      assert {:ok, _} =
```
The approach here is not correct.

You only have to create an empty package, then use the line_item you have set up to create a package item. You are already creating a package_item in `new_package` and using the attributes of that to create another package_item.

SagarKarwande commented on apps/snitch_core/test/data/model/package_item_test.exs:
```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
-                   product_id: line_item.product_id,
-                   package_id: make_package(context).id
-               })
+    test "with invalid attributes" do
```
Test statement is incorrect

SagarKarwande commented on apps/snitch_core/test/data/model/package_item_test.exs:
```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
 
-    test "with valid params", context do
-      %{line_items: [line_item]} = context
+    test "with invalid params", context do
```
Test statement is incorrect

arjun289 commented on apps/snitch_core/test/data/model/package_item_test.exs:
```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
-      shipping_method_id: sm.id
-    )
+
+  defp new_package(context) do
```
Break the function into two one should create a `package` another should create `package_item` and you should pass params which are required explicitly. You can make use of the context. The way `embedded_shipping_methods` is passed.

arjun289 commented on apps/snitch_core/test/data/model/package_item_test.exs:
```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
-               )
+    test "with a map", context do
+      %{package_item: package_item} = new_package(context)
+      map = %{quantity: package_item.quantity, delta: package_item.delta}
```
Use something which is unique for `package item`  e.g. package number. The keys that you have used are not unique for a package item. You have used quantity and delta, do you think these are unique for `package items`?

Jeshan7 committed:
[#163248592]
   - corrected the two test statements
   - made two function . First one creates the package and second one creates
     a package_item
   - In get/1 provided a unique value(:number) to retrieve that particular
     package_item
   - In test(line: 32) now first lin_item , product and package
       are created then package_item is created with valid parameters.

arjun289 reviewed (approved):
Looks good now.
